packages/contracts/src/personal/PersonalSpaceAdminPlugin.sol

```diff
@@ -48,16 +67,163 @@ contract PersonalSpaceAdminPlugin is PluginCloneable, ProposalUpgradeable {
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.grant,
+            (address(this), _newMember, MEMBER_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO revoke membership permission from the given address
+    /// @param _member The address that will no longer be a member
+    function submitRemoveMember(address _member) public auth(EDITOR_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.revoke,
+            (address(this), _member, MEMBER_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO revoke membership permission from the sender address
+    function leaveSpace() public auth(MEMBER_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.revoke,
+            (address(this), msg.sender, MEMBER_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO grant editor permission to the given address
+    /// @param _newEditor The address to grant editor permission to
+    function submitNewEditor(address _newEditor) public auth(EDITOR_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.grant,
+            (address(this), _newEditor, EDITOR_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO revoke editor permission from the given address
+    /// @param _editor The address that will no longer be an editor
+    function submitRemoveEditor(address _editor) public auth(EDITOR_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.revoke,
+            (address(this), _editor, EDITOR_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
```

Comment:
Should have a `EditorRemoved` event that emits the dao and the new editor's address